### ISSUE TITLE
请教一下memory unit

### ISSUE THREAD

**Issue by ouyx189** [X-PLUG/MobileAgent#26]
想问一下从代码上看好像planning agent和reflection agent的输入好像是没有memory参与的？
还有就是看你的demo运行速度很快，但实际我尝试调用api的时候一步就需要7，8秒的样子？

**Comment by junyangwang0410** [comment]
> 想问一下从代码上看好像planning agent和reflection agent的输入好像是没有memory参与的？ 还有就是看你的demo运行速度很快，但实际我尝试调用api的时候一步就需要7，8秒的样子？

planning agent和reflection agent的memory是从decision agent的输出获取的。

demo有加速处理，并且剪掉了等待相应的内容

**Comment by ouyx189** [comment]
拿reflection agent做例子,它的prompt输入是decision agent的summary,action。意思是memory也是从这两个获取的吗？

**Comment by junyangwang0410** [comment]
> 拿reflection agent做例子,它的prompt输入是decision agent的summary,action。意思是memory也是从这两个获取的吗？

是的。memory是decision agent做决策需要的，对于其他agent需要转换为agent所需的内容。例如reflection agent需要的操作意图，也就是decision agent的部分输出，是可能会用到memory中的内容的。
